File: codegen/results/v002/qwen3.5_4b_think/20260412_073207/typescript/004_json_filter/solution.ts
const fs = require('fs');

// Read the input file
const inputData = JSON.parse(fs.readFileSync('input/data.json', 'utf8'));

// Filter active records with age >= 30
const filteredData = inputData.filter(record => {
  return record.active === true && record.age >= 30;
});

// Sort by name ascending
filteredData.sort((a, b) => a.name.localeCompare(b.name));

// Output to stdout
console.log(JSON.stringify(filteredData));

File: codegen/results/v002/qwen3.5_4b_think/20260412_073207/python/004_json_filter/solution.py
import json

with open('input/data.json', 'r') as f:
    data = json.load(f)

result = [
    record for record in data
    if record['active'] and record['age'] >= 30
]

result.sort(key=lambda x: x['name'])

print(json.dumps(result))